demos/GMW_demo/1-out-of-n/IO.js

```diff
@@ -0,0 +1,59 @@
+/*
+ *  the IO object to be used for OT
+ *  to pass as the IO parameter for 1-out-of-N.
+ *
+ */
+// IO send
+const give = function ( op_id, jiff,session_id, tag, msg) {
+  /* give a message */
+  const createMSGObj=function (msg,op_id,tag,jiff) {
+    var four_opts={};
+    four_opts['opts']=msg;
+    four_opts['sender_id']=jiff.id;
+    four_opts['op_id']=op_id;
+    four_opts['tag']=tag;
+    return four_opts;
+  };
+  var temp=(op_id.split(':')[3]).split('-');
+  var to=parseInt(temp[2]); // sending msg to partyid
+  var from=parseInt(temp[1]);
+  var four_opts=createMSGObj(msg,op_id,tag,jiff) ;
+  four_opts = jiff.hooks.execute_array_hooks('beforeOperation', [jiff, 'open', four_opts], 2);
+  msg=JSON.stringify(four_opts);// jason object
+  var to_party=to;
+  if (jiff.id===to) {
+    to_party=from;
+  }
+  var recls=[];
+  recls.push(to_party);
+  jiff.emit('OT',recls,msg,true);
+  return ;
+};
+
+
+// IO receive
+const get = function (op_id,jiff,session_id, tag) {
+  /* get a message */
+  var temp=(op_id.split(':')[3]).split('-');
```

Comment:
same as above, this can be replaced with a call to `parse__OT_id()` or returning `[to_id, from_id]`  and then assign the local variable to the two pieces of the result.